pkg/getter/getter.go

```diff
@@ -85,6 +86,10 @@ func WithTLSClientConfig(certFile, keyFile, caFile string) Option {
 type Getter interface {
 	// Get file content by url string
 	Get(url string, options ...Option) (*bytes.Buffer, error)
+	// Retrieve the filename given a source URL
+	Filename(url *url.URL, version string) string
+	// Allow the getter to alter the download URL before retrieval
+	URL(u *url.URL, version string) (string, error)
```

Comment:
why should the getter need to alter the download URL prior to a call to `Get`? Shouldn't that be done during `Get`?

Reply:
That would require a breaking change to the `Get` function, as we need to pass the version in to automatically use that as the image tag if it was passed (https://github.com/helm/helm/pull/7613/files#diff-4e8b0b35b9905621c4d605f9681d9366R97). Happy to discuss alternatives here

Reply:
Do we need to consider adding a `func GetV2` or `type GetterV2`?

Reply:
or something like `func GetWithDetails` ?

Reply:
hm, we could do a GetterV2 or something 🤔 The problem we run into then is we need to introduce a bunch of case-based code elsewhere to check if you're using a v1 or v2 getter. We'd need to e.g pass something like a `GetterRef` that holds `Getter` and `GetterVersion` or something...

Reply:
Consider something like
```
type (
  Details struct {
      Content  *bytes.Buffer
      Filename string
      URL      string
  }

  Getter interface {
      Get(url string, options ...Option) (*bytes.Buffer, error)
      GetDetails(url string, options ...Option) (*Details, error)
  }
)
```

Then we just change anywhere using `Get` to use `GetWithDetails` - and use `details.Content` to replace the old bytes.Buffer used

Reply:
ah yeah we could do that. `GetWithDetails` would need to pass in the version string there, but that should be alright.

Reply:
I'm wondering if we can leave the same input args, and add version as an option?
```
func WithVersion(version string) Option {
	return func(opts *options) {
		opts.version = version
	}
}
```

Reply:
I did consider that, that would possibly allow using just the Get function too... I initially decided against it since the version didn't really feel like it was inline with other options; felt like purely an implementation detail.

I'll have a thorough look at how the Getter is implemented atm (and others too) to see if there's a better way. I do agree it doesn't feel very clean the way it's done now.